bundles/org.openhab.core.io.rest.sse/src/main/java/org/openhab/core/io/rest/sse/SseStatesResource.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.io.rest.sse;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Singleton;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.UriInfo;
+
+import org.glassfish.jersey.media.sse.EventOutput;
+import org.glassfish.jersey.media.sse.OutboundEvent;
+import org.glassfish.jersey.media.sse.SseFeature;
+import org.openhab.core.auth.Role;
+import org.openhab.core.io.rest.sse.internal.ItemStateChangesSseBroadcaster;
+import org.openhab.core.io.rest.sse.internal.SseStateEventOutput;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.items.events.ItemStateChangedEvent;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * SSE Resource for pushing selected item state updates to clients.
+ *
+ * @author Yannick Schaus - Initial contribution
+ */
+@Component(service = SseStatesResource.class)
+@Path(SseStatesResource.PATH_EVENTS)
+@RolesAllowed({ Role.USER })
+@Singleton
+@Api(value = SseStatesResource.PATH_EVENTS, hidden = true)
+public class SseStatesResource {
+
+    private final Logger logger = LoggerFactory.getLogger(SseStatesResource.class);
+
+    public static final String PATH_EVENTS = "events/states";
+
+    private static final String X_ACCEL_BUFFERING_HEADER = "X-Accel-Buffering";
+
+    private final ItemStateChangesSseBroadcaster broadcaster;
+
+    private final ExecutorService executorService;
+
+    private ItemRegistry itemRegistry;
+
+    @Reference
+    protected void setItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = itemRegistry;
+        this.broadcaster.setItemRegistry(itemRegistry);
+    }
+
+    protected void unsetItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = itemRegistry;
```

Comment:
for symmetry reasons it would be good to unset the item registry in the broadcaster here as well.